airflow_dbt_python/hooks/target.py

```diff
@@ -0,0 +1,351 @@
+                key = param
+                value = extra.get(key, None)
+
+            if value is None:
+                continue
+
+            dbt_details[key] = value
+
+        return dbt_details
+
+
+class DbtPostgresHook(DbtConnectionHook):
+    """A hook to interact with dbt using a Postgres connection."""
+
+    conn_type = "postgres"
+    hook_name = "dbt Postgres Hook"
+    conn_params = [
+        DbtConnectionParam("conn_type", "type", conn_type),
+        "host",
+        DbtConnectionParam("schema", default="public"),
+        DbtConnectionParam("login", "user"),
+        "password",
+        DbtConnectionParam("port", default=5432),
+    ]
+    conn_extra_params = [
+        DbtConnectionParam("dbname", "database", "postgres"),
+        "connect_timeout",
+        "role",
+        "search_path",
+        "keepalives_idle",
+        "sslmode",
+        "sslcert",
+        "sslkey",
+        "sslrootcert",
+        "retries",
+    ]
+
+    def get_dbt_details_from_connection(self, conn: Connection) -> dict[str, Any]:
+        """Extract dbt connection details from Airflow Connection.
+
+        dbt connection details may be present as Airflow Connection attributes or in the
+        Connection's extras. This class' conn_params and conn_extra_params will be used
+        to fetch required attributes from attributes and extras respectively. If
+        conn_extra_params is empty, we merge parameters with all extras.
+
+        Subclasses may override this class attributes to narrow down the connection
+        details for a specific dbt target (like Postgres, or Redshift).
+
+        Returns:
+            A dictionary of dbt connection details.
+        """
+        if "options" in conn.extra_dejson:
+            conn = copy(conn)
+            extra_dejson = conn.extra_dejson
+            options = extra_dejson.pop("options")
+            for k, v in re.findall(r"-c (\w+)=(.*)$", options):
+                extra_dejson[k] = v
+            conn.extra = json.dumps(extra_dejson)
+        return super().get_dbt_details_from_connection(conn)
+
+
+class DbtRedshiftHook(DbtPostgresHook):
+    """A hook to interact with dbt using a Redshift connection."""
+
+    conn_type = "redshift"
+    hook_name = "dbt Redshift Hook"
+    conn_extra_params = DbtPostgresHook.conn_extra_params + [
+        "method",
+        "cluster_id",
+        "iam_profile",
+        "autocreate",
+        "db_groups",
+        "ra3_node",
+        "connect_timeout",
+        "role",
+        "region",
+    ]
+
+
+class DbtSnowflakeHook(DbtConnectionHook):
+    """A hook to interact with dbt using a Snowflake connection."""
+
+    conn_type = "snowflake"
+    hook_name = "dbt Snowflake Hook"
+    conn_params = [
+        DbtConnectionParam("conn_type", "type", conn_type),
+        "host",
+        "schema",
+        DbtConnectionParam(
+            "login",
+            "user",
+            depends_on=lambda x: x.extra_dejson.get("authenticator", "") != "oauth",
```

Comment:
My comment on replacing these lambdas is still relevant, but it was more of a nitpick.

Reply:
Closing this in favor of the aforementioned comment.